mypy/nodes.py

```diff
@@ -2658,7 +2658,13 @@ class is generic then it will be a type constructor of higher kind.
     # even though it's not a subclass in Python.  The non-standard
     # `@_promote` decorator introduces this, and there are also
     # several builtin examples, in particular `int` -> `float`.
-    _promote: Optional["mypy.types.Type"]
+    _promote: List["mypy.types.Type"]
+
+    # This is used for promoting native integer types such as 'i64' to
+    # 'int'. (_promote is used for the other direction.) This only
+    # supports one-step promotions (e.g., C -> D, not C -> D -> E),
```

Comment:
Yes, this may be confusing. If we'd allow this, then `join(i32, float)` would have to be `float`, I think (since we can't allow assignment from `float` to `i32`), which would be inconsistent with `join(i32, int)`, which is `i32` (due to "stickiness"). So something will behave against expectations, and my thinking was that it's more important to have easy interop with `int` than `float`.